ARCHITECTURE.md

```diff
@@ -0,0 +1,56 @@
+# PrestoDB: Mission and Architecture
+
+## Introduction
+
+PrestoDB is an open-source distributed SQL query engine designed for fast analytic queries against data sources of all sizes, ranging from gigabytes to petabytes. It was originally developed by Facebook and later made available to the broader community. It is governed under the auspices of the Presto Foundation, a member of the Linux Foundation.
+
+## Mission
+
+### For users
+
+The primary mission of PrestoDB is to enable efficient and high-speed data processing for analytics and batch at scale. It aims to provide a single, unified query system that can access and process data stored in various formats and storage systems. Key aspects of its mission include:
+
+1. **High Performance:** Presto strives to exceed data warehouse performance in data lakes. To achieve this, Presto works to achieve vertical integration over modular portions of the stack.
+2. **Scalability:** Presto is used at extremely large data lakes across the world. As such, it aims for high degrees of scalability.
+3. **Connectivity and Integration:** Presto is designed to be flexible and adaptable to changes in infrastructure. This includes a highly customizable plugin infrastructure, including pluggable storage connectors.
+4. **User-Friendly:** Presto strives to be simple enough for a user to spin up and quickly test out, using familiar ANSI style SQL syntax, sensible default values and an optimizer that produces efficient query plans.
```

Comment:
It is not ANSI SQL, and you are correct that nothing is ANSI SQL.  But we strive for ANSI compliant syntax.  We reference the spec and defer to the spec.  By using the term "ANSI style", I am saying that that is what we strive for, without saying we *are* ANSI SQL.  How would you word this?  It's important to us as a project that we defer to the spec, so it needs to stay in this document somehow.

Reply:
I still wouldn't use the word ANSI at all, but if you must add quotes around "ANSI style" to weaken it a bit.

Reply:
```suggestion
4. **User-Friendly:** Presto strives to be simple enough for a user to spin up and quickly test out, using SQL syntax that strives towards ANSI compliance, sensible default values, and an optimizer that produces efficient query plans.
```
I don't recommend quoting "ANSI style" in the document. Air or scare quotes emphasizing a word or phrase to convey irony or skepticism should be avoided because the meaning of the quoted words in the documents becomes vague. Scare quotes tell the reader that the words here aren't exactly what is meant but the writer hasn't clearly explained what the writer means to say, only that the quoted words aren't quite it. 

For one example of discussion, see heading 4 (Set apart a word to show irony, sarcasm, or skepticism (scare quotes)) in [Quotation Marks: Rules and Examples](https://www.grammarly.com/blog/quotation-marks/).  

My suggested change "using SQL syntax that strives toward ANSI compliance" adapts part of @tdcmeehan's explanation in [his comment](https://github.com/prestodb/presto/pull/22356/files#r1546801463), softening the implied claim by emphasizing the goal and that Presto SQL isn't there yet.

Reply:
"Scare quotes" I'll have to remember that term in the future, but it is exactly what I wanted to convey.

Let me put it this way: if PrestoSQL is ANSI style SQL, with or without scare quotes, what is a dialect that is not ANSI style? What can a reader practically infer by knowing that PrestoSQL is ANSI style SQL as compared to simply knowing that it's SQL?

Reply:
@elharo there are varying degrees of compliance toward the spec.  As I've explained, our intention is to hew toward the spec where possible, whereas other systems do not have any such intention.  What do you think of @steveburnett's writing suggestion above, do you think that conveys this message?

Reply:
@elharo a dialect that is not ANSI style is Spark SQL (prior to Spark 3.x).  It is called SQL, but differs in many ways from the ANSI standard, to the extent that there is now an ANSI mode in 3.x that differs from the old, non-ANSI mode prior.